editor/editor_export.cpp

```diff
@@ -1658,8 +1658,9 @@ Ref<Texture> EditorExportPlatformPC::get_logo() const {
 
 bool EditorExportPlatformPC::can_export(const Ref<EditorExportPreset> &p_preset, String &r_error, bool &r_missing_templates) const {
 	String err;
-	bool valid = false;
+	bool valid = true;
```

Comment:
Why default to `true`?

Reply:
The `valid` variable is only updated to `true` when the templates are valid. Since for the Android port, we're skipping the templates' checks, it's never updated to `true`, and thus always returns `false`.

For non-Android port, the initial value for `valid` is always overwritten, so there's no risk of affecting the logic for the other platforms.

Reply:
Yeah but it seems to be a workaround here. This method checks if export templates are valid. If they're not valid on PC, it should behave the same as if they're not valid on Android. With this change it will be `true` on Android and `false` on PC, and the latter seems more accurate to me.

Sounds like what you want to do is just enable PCK export, and the problem here is that the logic to enable it is coupled to the availability of templates. Which is partially true at least for Mono (BCL needs to be included in the PCK), but can also not be a deal-breaker for simple projects.

So what's needed IMO is a bigger refactor that decouples PCK export check from binary export check.

Reply:
> This method checks if export templates are valid. If they're not valid on PC, it should behave the same as if they're not valid on Android. With this change it will be true on Android and false on PC, and the latter seems more accurate to me.

The method unfortunately is a catch-all for export templates validity checks, and project configuration validity checks.
For a full binary export, both checks are valid. However for a `pck/zip` only export, the export templates validity checks are irrelevant.
So given we'll only (initially) support `pck/zip` export for the Android editor, the export templates validity checks are being disabled. So yes in this scenario, if the project configuration validity checks fail, they fail both on Android and Windows. If the export templates validity checks fail, they only fail on Windows but not on Android.

> Which is partially true at least for Mono (BCL needs to be included in the PCK), but can also not be a deal-breaker for simple projects.

The flow I'm imagining is that a user would do development and testing using the Android editor port, then via export of a zip or pck file, remote versioning or file syncing, continue and finish the release process on a fully featured platform. Would the BCL need to be included for that workflow to work?

 >  Or if a temporary hack is needed for Android, it can simply be not to call can_export since your changes basically makes it a no-op.

The `can_export` call is still needed for the project configuration validity checks. This particular method is overwritten by the `EditorExportPlatformWindows::can_export` method which uses this parent logic for its template validation logic, so the short-circuit for Android is still needed.

> So what's needed IMO is a bigger refactor that decouples PCK export check from binary export check.

Agree this would be a good approach as well. Splitting `can_export` into a `validate_export_templates` and a `validate_project_configuration` methods. We can then skip the call to the `validate_export_templates` for the Android editor.